cmd/mattermost/commands/plugin.go

```diff
@@ -0,0 +1,181 @@
+	"errors"
+	"os"
+
+	"github.com/spf13/cobra"
+)
+
+var PluginCmd = &cobra.Command{
+	Use:   "plugin",
+	Short: "Management of plugins",
+}
+
+var PluginCreateCmd = &cobra.Command{
+	Use:     "add",
+	Short:   "Add a plugin",
+	Long:    "Add a plugin or multiple plugins",
+	Example: `  plugin add hovercardexample.tar.gz pluginexample.tar.gz`,
+	RunE:    pluginCreateCmdF,
+}
+
+var PluginDeleteCmd = &cobra.Command{
+	Use:     "delete",
+	Short:   "Delete a plugin",
+	Long:    "Delete a plugin",
+	Example: `  plugin delete hovercardexample`,
+	RunE:    pluginDeleteCmdF,
+}
+
+var PluginEnableCmd = &cobra.Command{
+	Use:     "enable",
+	Short:   "Enable a plugin",
+	Long:    "Enable a plugin or multiple plugins",
+	Example: `  plugin enable hovercardexample.tar.gz pluginexample.tar.gz`,
+	RunE:    pluginEnableCmdF,
+}
+
+var PluginDisableCmd = &cobra.Command{
+	Use:     "disable",
+	Short:   "Disable a plugin",
+	Long:    "Disable a plugin",
+	Example: `  plugin disable hovercardexample`,
+	RunE:    pluginDisableCmdF,
+}
+
+var PluginListCmd = &cobra.Command{
+	Use:     "list",
+	Short:   "List plugins",
+	Long:    "List all active and inactive plugins",
+	Example: `  plugin list`,
+	RunE:    pluginListCmdF,
+}
+
+func init() {
+	PluginCmd.AddCommand(
+		PluginCreateCmd,
+		PluginDeleteCmd,
+		PluginEnableCmd,
+		PluginDisableCmd,
+		PluginListCmd,
+	)
+	RootCmd.AddCommand(PluginCmd)
+}
+
+func pluginCreateCmdF(command *cobra.Command, args []string) error {
+	a, err := InitDBCommandContextCobra(command)
+	if err != nil {
+		return err
+	}
+	defer a.Shutdown()
+
+	if len(args) < 1 {
+		return errors.New("Expected at least one argument. See help text for details.")
+	}
+
+	for i, plugin := range args {
+		fileReader, err := os.Open(plugin)
+		if err != nil {
+			return err
+		}
+
+		if _, err := a.InstallPlugin(fileReader); err != nil {
+			return errors.New("Unable to create plugin: " + args[i] + ". Error: " + err.Error())
+		}
+		fileReader.Close()
+	}
+
+	CommandPrettyPrintln("Created plugin(s)")
+
+	return nil
+}
+
+func pluginDeleteCmdF(command *cobra.Command, args []string) error {
+	a, err := InitDBCommandContextCobra(command)
+	if err != nil {
+		return err
+	}
+	defer a.Shutdown()
+
+	if len(args) < 1 {
+		return errors.New("Expected at least one argument. See help text for details.")
+	}
+
+	if err := a.RemovePlugin(args[0]); err != nil {
+		return errors.New("Unable to delete plugin: " + args[0] + ". Error: " + err.Error())
+	}
+
+	CommandPrettyPrintln("Deleted plugin")
+
+	return nil
+}
+
+func pluginEnableCmdF(command *cobra.Command, args []string) error {
+	a, err := InitDBCommandContextCobra(command)
+	if err != nil {
+		return err
+	}
+	defer a.Shutdown()
+
+	if len(args) < 1 {
+		return errors.New("Expected at least one argument. See help text for details.")
+	}
+
+	for i, plugin := range args {
+		if err := a.EnablePlugin(plugin); err != nil {
+			return errors.New("Unable to enable plugin: " + args[i] + ". Error: " + err.Error())
```

Comment:
Just something to think about (Not a stopper at all), which behavior do we want here?
Now the behavior is "Enable plugins until you find someone which is not possible to be enabled". I would expect something like "Try to enable this plugins, if someone fails, let me know it, but follow enabling the rest of the plugins".

Reply:
I like the second approach of trying the rest of the plugin. Instead of returning `errors.New()` message, could we do something like this?

```go
if err := a.EnablePlugin(plugin); err != nil {
    CommandPrintErrorln("Unable to enable plugin: " + args[i] + ". Error: " + err.Error())
}
```

Reply:
Yes, and I would add some variable to track if there was any error and show something different at the end. Maybe something like a list of correctly enabled plugins and the list of failed.